sources/api/apiserver/src/server/controller.rs

```diff
@@ -420,21 +498,91 @@ pub(crate) fn get_metadata_for_data_keys<D: DataStore, S: AsRef<str>>(
     Ok(result)
 }
 
-/// Gets the value of a metadata key everywhere it's found in the data store.  Returns a mapping
-/// of data key to the metadata value associated with the requested key.
-pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+// Sort the given Hashmap in topological order
+fn get_sorted_metadata(
+    metadata: HashMap<Key, HashMap<Key, String>>,
+) -> Vec<(Key, HashMap<Key, String>)> {
+    let mut metadata_sorted: Vec<_> = metadata.into_iter().collect();
+    metadata_sorted.sort_by(|(k1, _), (k2, _)| {
+        k1.segments()
+            .len()
+            .cmp(&k2.segments().len())
+            .then_with(|| k1.name().cmp(k2.name()))
+    });
+    metadata_sorted.to_vec()
+}
+
+// Find the successor of the given setting and add the metadata
+// for the child at given depth in setting generator
+fn get_setting_generator_for_child<D: DataStore>(
```

Comment:
Similar naming feedback regarding the word `get`. Also, be sure to use docstring comments for function documentation.

```suggestion
/// Settings generators can be dynamically applied to a set of settings.
///
/// This function resolves a dynamically expanded generator into a finite set of
/// distinct settings generators.
fn expand_setting_generator<D: DataStore>(
```